galsim/table.py

```diff
@@ -165,18 +179,22 @@ def __call__(self, x):
             if dimen > 2:
                 raise ValueError("Arrays with dimension larger than 2 not allowed!")
             elif dimen == 2:
-                f = np.zeros_like(x)
-                for i in xrange(x.shape[0]):
-                    f[i,:] = np.fromiter((self.table(float(q)) for q in x[i,:]), dtype='float')
+                f = np.empty_like(x.flatten(), dtype=float)
```

Comment:
Not sure if it will make much difference, but I think .ravel() is usually faster than .flatten()  (yields a view instead of copying).